mongo/gridfs/bucket.go

```diff
@@ -617,31 +596,13 @@ func (b *Bucket) parseUploadOptions(opts ...*options.UploadOptions) (*Upload, er
 		if err != nil {
 			return nil, err
 		}
-		doc, err := bsonx.ReadDoc(raw)
-		if err != nil {
-			return nil, err
+		doc := &bson.D{}
```

Comment:
[opt] Usually I see `var doc bson.D` and then `&doc` in the call to `UnmarshalWithRegistry`. Then you could avoid the dereference with `upload.metadata = doc`.